tla-types/src/main/scala/at/forsyte/apalache/tla/typecmp/subbuilder/LeafBuilder.scala

```diff
@@ -0,0 +1,65 @@
+package at.forsyte.apalache.tla.typecmp.subbuilder
+
+import at.forsyte.apalache.tla.lir._
+import at.forsyte.apalache.tla.typecmp._
+import at.forsyte.apalache.tla.typecmp.raw.RawLeafBuilder
+import scalaz.Scalaz._
+import scalaz._
+
+/**
+ * Builder for names and literals (IR tree leaves)
+ *
+ * @author
+ *   Jure Kukovec
+ */
+trait LeafBuilder extends RawLeafBuilder {
```

Comment:
```suggestion
trait LiteralAndNameBuilder extends RawLeafBuilder {
```

Names should reflect the use, not the implementation or another unnecessary detail.